apps/src/lab2/levelEditors/aichatSettings/EditAichatSettings.tsx

```diff
@@ -46,8 +47,14 @@ function sanitizeSettings(settings: LevelAichatSettings) {
     DEFAULT_VISIBILITIES
   );
 
+  // Ensure that only valid model IDs are included.
+  const filteredModelIds = (settings.availableModelIds || []).filter(id =>
+    modelDescriptions.map(model => model.id).includes(id)
+  );
```

Comment:
non-blocking, but I think this'd be a little better with a `some` call.

```
const filteredModelIds = (settings.availableModelIds || []).filter(id =>
  modelDescriptions.some(model => model.id === id)
);
```

It's O(n^2) instead of O(n^3) (not that it really matters with such small loops), and I lean towards it being a little easier to read.

OTOH, I also acknowledge it's not _that_ much easier to read, and maybe it loses something in legibility going to a `some` call instead of a more clear sounding `includes` call. 🤷 

I dunno. Dealer's choice.

It could also be an external set:
```
const validModelIds = new Set(modelDescriptions.map(model => model.id));
const filteredModelIds = (settings.availableModelIds || []).filter(id => validModelIds.includes(id) );
```

That's clearer and faster (O(2n)), but it's also using an ugly external variable for just one thing.

Anyway, I'm just rambling.

Reply:
Ah yep, good call, I'll update this